charts/node/templates/statefulset.yaml

```diff
@@ -665,24 +662,23 @@ spec:
             {{- toYaml . | nindent 12 }}
             {{- end}}
           ports:
-            {{- if .Values.node.legacyRpcFlags }}
-            - containerPort: 9933
+            {{- if $.Values.node.legacyRpcFlags }}
+            - containerPort: {{ $.Values.node.perNodeServices.apiService.httpPort | int }}
               name: http-rpc
-              protocol: TCP
-            {{- end }}
-            - containerPort: 9944
+            - containerPort: {{ $.Values.node.perNodeServices.apiService.wsPort | int }}
```

Comment:
This may break old deployments.  
If `legacyRpcFlags` is `true` the  `--rpc-port` `--ws-port` flags are not set. The node will be listen port  `9933` and `9944`. Setting `.apiService.wsPort` or `.apiService.httpPort` will not affect the node but will break the service.

Reply:
I'm reverting to hardcoded 9933/9944 for legacy rpc flags in https://github.com/paritytech/helm-charts/pull/241/commits/9e9f48f6a21594a0080161bdbe2202b1047db40d